### DISCUSSION TITLE
Does C# Support CSP (Communication Sequential Process)?

### DISCUSSION THREAD

**Discussion by sajjadsaharkhan** [dotnet/runtime#98747] [Q&A]
In golang we have a concept that allows to transfer data between process concurrently or In parallel.
[Communicating Sequential Processes](https://en.wikipedia.org/wiki/Communicating_sequential_processes)

In C# we have a data type called [Channel](https://learn.microsoft.com/en-us/dotnet/core/extensions/channels) that allows us to transfer data between some `Task`  is it really CSP?

**Comment by huoyaoyuan** [comment]
There was built-in remoting in .NET framework, but then abandoned due to security risk: the remote peer can be arbitrary and malicious.

**Comment by YousefAK009** [comment]
I am looking for this too
